test/conformance/enqueue/urEnqueueKernelLaunch.cpp

```diff
@@ -262,3 +262,72 @@ TEST_P(urEnqueueKernelLaunchMultiDeviceTest, KernelLaunchReadDifferentQueues) {
         ASSERT_EQ(val, output) << "Result on queue " << i << " did not match!";
     }
 }
+
+struct urEnqueueKernelLaunchUSMLinkedList : uur::urBaseKernelExecutionTest {
+    struct Node {
+        Node() : next(nullptr), num(0xDEADBEEF) {}
+
+        Node *next;
+        uint32_t num;
+    };
+
+    void SetUp() override {
+        program_name = "usm_ll";
+        UUR_RETURN_ON_FATAL_FAILURE(urBaseKernelExecutionTest::SetUp());
+
+        Build();
+    }
+
+    void TearDown() override {
+        auto *list_cur = list_head;
+        while (list_cur) {
+            auto *list_next = list_cur->next;
+            ASSERT_SUCCESS(urUSMFree(context, list_head));
+            list_cur = list_next;
+        }
+
+        UUR_RETURN_ON_FATAL_FAILURE(urBaseKernelExecutionTest::TearDown());
+    }
+
+    size_t global_size = 1;
+    size_t global_offset = 0;
+    Node *list_head = nullptr;
+    const int num_nodes = 4;
+};
+
+UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urEnqueueKernelLaunchUSMLinkedList);
+
+TEST_P(urEnqueueKernelLaunchUSMLinkedList, Success) {
+    // Build linked list with USM allocations
+    ASSERT_SUCCESS(urUSMSharedAlloc(context, device, nullptr, nullptr,
```

Comment:
could you add a case that uses a pool with the allocations as well?

Reply:
Done now, I ended up discovering a bug in the original test which is fixed too